blockchain/chain/src/store/chain_store.rs

```diff
@@ -106,4 +107,53 @@ impl<'a> ChainStore<'a> {
     pub fn blockstore(&self) -> &Blockstore {
         &self.db
     }
+    /// Returns Tipset from key-value store from provided keys
+    pub fn tipset(&self, keys: &[Cid]) -> Result<Tipset, Error> {
+        let mut block_headers = Vec::new();
+        for k in keys {
+            let raw_header = self.db.read(k.key())?;
+            if let Some(ref x) = raw_header {
```

Comment:
```suggestion
            if let Some(x) = raw_header {
```
nit, rust will dereference it anyway but there is no need here